src/Workspaces/Core/Portable/FindSymbols/SymbolFinder.cs

```diff
@@ -15,7 +15,11 @@ namespace Microsoft.CodeAnalysis.FindSymbols
     public static partial class SymbolFinder
     {
         /// <summary>
-        /// Finds the symbol that is associated with a position in the text of a document.
+        /// Finds the symbol that is associated with a position in the text of a document. If 
+        /// multiple symbols are associated with the token (e.g. implicitly named anonymous type
```

Comment:
@dpoeschl has said pretty much everything I hated about the "return a list of symbols" API. We originally had internal helpers that returned the list of symbols, and in practice a lot of callers were just calling .FirstOrDefault(). It wasn't even clear for most features what they could do with all the symbols, or for some cases why all the symbols were there. If you have a token that's both a declaration and a reference, what should the order in the list be? What would it mean to change that order? Any time I had to touch that code it was a bug farm because the order of the list was tuned to whoever last changed the order for their pet feature.

In practice, my assumption was that the user expects that if they invoke a half dozen features on the same point in a file, that the symbol that we act on should be the same. Yes, some features _might_ be able to do something special, but this is the API that exists to cover the majority of the cases, and is intended to be the "easy" path into Roslyn. Hence we went with the API that gives you "the" symbol, where "the" symbol is the one chosen via fiat.